internal/cmd/plugin/destroy/cmd.go

```diff
@@ -42,12 +46,60 @@ func NewCmd() *cobra.Command {
 			}
 
 			keepPVC, _ := cmd.Flags().GetBool("keep-pvc")
+			force, _ := cmd.Flags().GetBool("force")
+			if !force {
+				if err := ensureOtherRunningInstances(ctx, clusterName, node); err != nil {
+					return err
+				}
+			}
 			return Destroy(ctx, clusterName, node, keepPVC)
 		},
 	}
 
 	destroyCmd.Flags().BoolP("keep-pvc", "k", false,
 		"Keep the PVC but detach it from instance")
 
+	destroyCmd.Flags().BoolP("force", "f", false,
+		"Force the deletion, even if it is the last remaining instance")
+
 	return destroyCmd
 }
+
+func ensureOtherRunningInstances(ctx context.Context, clusterName string, node string) error {
+	// List all pods for the cluster
+	var podList corev1.PodList
+	if err := plugin.Client.List(ctx, &podList, client.InNamespace(plugin.Namespace), client.MatchingLabels{
+		utils.ClusterLabelName: clusterName,
+		utils.PodRoleLabelName: string(utils.PodRoleInstance),
+	}); err != nil {
+		return fmt.Errorf("error listing pods for cluster %s: %v", clusterName, err)
+	}
+
+	var otherRunningInstancesCount int
+	for _, pod := range podList.Items {
+		if pod.Name == node {
+			continue
+		}
+
+		if utils.IsPodReady(pod) && utils.IsPodActive(pod) {
+			otherRunningInstancesCount++
+		}
+	}
+
+	if otherRunningInstancesCount > 0 {
+		return nil
+	}
+
+	fmt.Printf(
+		"WARNING: No running instances remain. Are you sure you want to attempt destroying the instance? [y/N]: ",
+	)
+	var response string
+	if _, err := fmt.Scanln(&response); err != nil {
+		return err
+	}
+	if strings.ToLower(response) != "y" {
+		return fmt.Errorf("operation aborted by the user")
+	}
```

Comment:
If we do this, `kubectl cnpg destroy` will no longer be usable from a shell script without using the `--force` flag.

We could avoid being interactive using a strategy similar to what `tar` and `rm` do.
That would mean:

1. if we'll be deleting the last PVC group
    1. with no `force` flag, we emit a message and refuse to do that 
    2. with the `force` flag, we delete it
3. if we're not deleting the last PVC group, do it

What to you think of that?